drivers/fuel_gauge/sy24561/emul_sy24561.c

```diff
@@ -120,13 +121,13 @@ static int sy24561_emul_transfer_i2c(const struct emul *target, struct i2c_msg *
 
 		uint16_t val;
 
-		ret = emul_sy24561_reg_read(target, *msgs->buf, &val);
+		ret = emul_sy24561_reg_read(target, *msgs[0].buf, &val);
 		if (ret) {
 			LOG_ERR("emul_sy24561_reg_read returned %d", ret);
 			return ret;
 		}
 
-		sys_put_be16(val, msgs->buf);
+		sys_put_be16(val, msgs[1].buf);
```

Comment:
This is a functional change I consider to improve the emulation. @franckduriez I would really appreciate a review from your side to confirm my fixes